scripts/vast-completions.bash

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+_vast_completions()
+{
+  completion_words=("${COMP_WORDS[@]}")
+  if [ "${COMP_WORDS[@]: -1}" == "--" ]; then
+    # We need to special-case '--', since that has special
+    # meaning, e.g.
+    #
+    #    vast pivot --asdf help -> displays error message/help text
+    #    vast pivot -- help     -> runs pivot command
+    #
+    # Technically we should scan the whole COMP_WORDS array
+    # for '--', but that issue should come up rarely, wheras
+    # a '--' at the end happens all the time as the result of
+    # a previous autocompletion.
+    completion_words[$((${#COMP_WORDS}-1))]="-"
+  fi
+  tmpfile=$(mktemp)
```

Comment:
What does using a temporary file achieve here? We should not need to write anything to disk here, this is just a completion script, and it should work on a read-only filesystem as well.

Reply:
It's doing the conversion `string with newlines` -> `bash array`, where each array element should be one line of the input string. Please share if you know of a better way to do this :)

Reply:
```bash
IFS=$'\n' read -rd '' -a bash_array <<<"$string_with_newlines"
```

This should work and is portable between zsh 4+ and bash 3.2+

Reply:
This is portable between zsh and bash.

```bash
IFS=$'\n' y=($(<command that outputs string with newlines>))
```

Reply:
Nice, added an adapted version (for some reason, without the `$SAVEIFS` workaround both this version and one with `local IFS=...` would break the subsequent `compgen` call; maybe this is a bash bug because IFS is a special built-in variable)